client/my-sites/stats/stats-purchase/stats-purchase-tier-upgrade-slider.tsx

```diff
@@ -0,0 +1,73 @@
+import { PricingSlider } from '@automattic/components';
+import classNames from 'classnames';
+import { translate } from 'i18n-calypso';
+import React, { useState } from 'react';
+import useAvailableUpgradeTiers from './stats-purchase-tier-upgrade-slider-utils';
+
+import './stats-purchase-tier-upgrade-slider.scss';
+
+function getLocalizedStrings() {
+	const limits = translate( 'Monthly site views limit', {
+		comment: 'Heading for Stats Upgrade slider. The monthly views limit.',
+	} );
+	const price = translate( 'You will pay', {
+		comment: 'Heading for Stats Upgrade slider. The monthly price.',
+	} );
+	const strategy = translate( 'Price per month, billed yearly', {
+		comment: 'Stats Upgrade slider message. The billing strategy.',
+	} );
+
+	return {
+		limits,
+		price,
+		strategy,
+	};
+}
+
+type TierUpgradeSliderProps = {
+	className?: string;
+};
+
+function TierUpgradeSlider( { className }: TierUpgradeSliderProps ) {
+	const componentClassNames = classNames( 'stats-tier-upgrade-slider', className ? className : '' );
+
+	// Get the plan details.
+	const plans = useAvailableUpgradeTiers();
+
+	// Slider state.
+	const [ currentPlanIndex, setCurrentPlanIndex ] = useState( 0 );
+	const sliderMin = 0;
+	const sliderMax = plans.length - 1;
+
+	const handleSliderChange = ( value: number ) => {
+		setCurrentPlanIndex( value );
+	};
+
+	const localizedStrings = getLocalizedStrings();
```

Comment:
I think it's better to put the object here if you want to use the object. The function doesn't do anything to the strings (apart from putting it into an object), so we don't have to separate it. It's also easier to read when the strings are here rather than checking what's happening in the function.

I'm not sure we need to bundle the strings into an object. It would be more transparent if the translation stayed in h2.

Reply:
Added to parent issue as a new task. I feel like this is a matter of style though. For me personally, I prefer to separate the string translation calls from the JSX and they are noisy. I don't have strong feelings about wrapping them in a function though.

Reply:
It does look clearer this way; however the problem with use `translate` outside of components is that, if user switches languages or the translation file is loaded after rendering, the component wouldn't reflect the changes if that makes sense. As a result, I would recommend always use hook - `useTranslate` 😃

Reply:
That said, I think you should still be able to create a hook to translate strings with `useTranslate` like your pattern, for example:

```JavaScript
function useTranslationStrings() {
    const translate = useTranslate();
    // useMemo is optional here, as the translation process is pretty light.
    return useMemo( () => ( { message: translate( 'This is a message' ) } ), [ translate ] );
}
```

PS: I didn't test it. You might need to tweak a bit to make it work 😄

Reply:
Thanks for this explanation, Jasper. So in the case of `translate` vs `useTranslate` it's not just a matter of style. I'll note this in the list of follow-up tasks. 👍